### DISCUSSION TITLE
Some of the reactflow instance functions aren't working for some reason

### DISCUSSION THREAD

**Discussion by slechanii** [xyflow/xyflow#2255] [General]
Hello,

I'm currently using ReactFlow V10 and getting the instance using `const flowInstance = useReactFlow();`
I have been using some of the instance functions like` getNodes`, `getEdges`, `project` successfully so far and I would now like to use functions that will change the graph in some way (`getViewport`, `setViewport`, `zoomIn`) but none of these are working.
`setViewport` and `zoomIn`/`zoomOut` does nothing while `getViewport` always returns the same things (`{ x: 0, y: 0, zoom: 1 }`) no matter how much I move / zoom.

It really looks like the viewport of the instance is not really connected to the viewport of the graph because everything is working except functions that will interact with the viewport.

Does someone have any idea what could cause viewport related functions of reactflow instance to not work ?

Thanks for the help.

**Comment by arickert** [comment]
I am having the same issue!

**Reply by moklick** [reply]
We are not support v10 anymore. Please update to the latest version. 

Is it possible that you are using `zoomIn` or `flowInstance` inside a `useEffect` or `useCallback` and not passing it to the dependency array?

**Reply by infinity-atom42** [reply]
After more than 1 hour, now I understand why people hate React

Thx btw
